lib/rubygems/package.rb

```diff
@@ -200,7 +200,11 @@ def add_contents tar # :nodoc:
 
   def add_files tar # :nodoc:
     @spec.files.each do |file|
-      stat = File.stat file
+      stat = File.lstat file
+
+      if stat.symlink?
+        tar.add_symlink file, File.readlink(file), stat.mode
+      end
```

Comment:
For this chunk please create an `add_files_symlink` test mirroring the [`add_files`](https://github.com/rubygems/rubygems/blob/master/test/rubygems/test_gem_package.rb#L106) test.